### PR TITLE
[CI] Add macos C++ build to ci.yml

### PR DESCRIPTION
**Pull request by PragmaTwice**
macOS C++ build is added to CI workflow in this PR.

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -148,9 +132,13 @@ jobs:
       - uses: actions/checkout@v3
       - uses: dtolnay/rust-toolchain@nightly
       - run: ./ci/run_ci.sh rust
+
   cpp:
     name: C++ CI
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```
Comment (chaokunyang) [inline .github/workflows/ci.yml:140]: Does this include bont Mac Intel and Mac silicon?

Comment (PragmaTwice) [reply .github/workflows/ci.yml:140]: No. Currently it's x86.

Comment (PragmaTwice) [reply .github/workflows/ci.yml:140]: Actually it's a hard problem for arm64-based GitHub Action runners (it is currently not supported by GitHub official runners, but supported only in self-hosted runners). cc @tisonkun may have some information.

Comment (chaokunyang) [reply .github/workflows/ci.yml:140]: We never release fury python for Mac M1/M2, seems it's not easy to automate it. It's beyoud the scope of this PR, we can discuss it in #812

**Pull request merged**